### DISCUSSION TITLE
Content Editor Components - Visual Picker

### DISCUSSION THREAD

**Discussion by ceyerkes** [twilio-labs/paste#2884] [General] [answered]
Hellooo! I took a stab at the visual picker, button/card hybrid for the content editor: 

![Screen Shot 2022-12-14 at 8 56 03 AM](https://user-images.githubusercontent.com/87141551/207644467-ad62afc9-48ef-44d4-ace2-18ca6b6a909e.png)

Default, hover, active and focus mirror current button states, and I added a couple select state options as well. Hover states in the editors of MC, Unlayer, and MailChimp are _terrible._ You can barely tell what you're hovering on, and I think incorporating these button states will improve accessibility dramatically. All 3 of these editors only have a default state, faint hover state and drag state. Still need to figure out what we want the button to look like mid-drag.

This is current MC, and what i'm trying to replicate in Paste:

![Screen Shot 2022-12-13 at 7 38 07 PM](https://user-images.githubusercontent.com/87141551/207491642-2af3806c-6678-4e3c-a718-ac11ae4e0877.png)

The only two use cases I have for the select state in the content editor so far is when a user is selecting a column arrangement in MC:

![Screen Shot 2022-12-13 at 7 34 07 PM](https://user-images.githubusercontent.com/87141551/207491154-0689a725-cf65-4eb1-86b7-b8e4b9f2e5ab.png)

And also when a user initially selects an editing path:

![Screen Shot 2022-12-14 at 8 57 25 AM](https://user-images.githubusercontent.com/87141551/207644839-f44ee77a-33c0-4555-b900-d0997a174c81.png)

Please let me know if I can provide any additional context, and/or if office hours makes sense to discuss further. Thanks!

**Comment by nkrantz** [comment]
Hi @ceyerkes - these are looking incredible! Awesome work here, especially with all those state designs. The team has seen your discussions and we're excited to work more with you in the new year. 

We'll be out of office next week during the company break, and will be slower to respond to GitHub Discussions and other questions through the new year. We'll be back to our regular response times on January 2, 2023. Talk soon!

**Comment by ceyerkes** [comment, marked as answer]
Hello! I incorporated feedback from @brannytempy before the break, and wanted to keep this thread updated. 

I added:
- drag handles
- elevation 
- increased size of triangle checkmark area
- changed checkmark icon to action/accept
- drag state with increased elevation
- radio button version

![Screenshot 2023-01-03 at 2 44 38 PM](https://user-images.githubusercontent.com/87141551/210446425-5837e810-0495-44ff-a4ba-3bb9a7a555c4.png)

**Reply by ceyerkes** [reply]
Also - not sure how i deleted the comment i made before this one, but i can't get it back 😢

**Reply by TheSisb** [reply]
Thanks for the update! These look great! 
Unfortunately I don't think there is a way to recover a deleted post on Github Discussions :(

**Reply by kelseycavitt** [reply]
@ceyerkes this is looking great! Could you share the Figma link to see these ideas in context as well before OH tomorrow? Also sorry about your lost comment! Excited to chat more in office hours!

**Reply by ceyerkes** [reply]
No prob! [Figma link](https://www.figma.com/file/AThoZNMoCXViDddWDbKb0d/Content-Editor?node-id=519%3A211051&t=jV0PqBLlWWxrp2Rp-1)

**Reply by kelseycavitt** [reply]
We really enjoyed getting to work through your visual picker concepts last week in [Office Hours](https://www.loom.com/share/c74a3dd40081418fb4fab16ab274de9c)!

Here’s a summary of our discussion as well as some guidance:

- The visual picker is on our roadmap, but won’t be picked up at this time, as our quarter has been prioritized already.
- For the mutually exclusive options, like the column selector, you may not need a radio button within the card itself and can use your selected state idea (within the Figma doc). 
- For the draggable variant, it does not need a selected state, as they are drag and drop, but they do need a focus state. When you end up implementing the draggable elements, please ensure the component is keyboard accessible and is read properly by a screen reader. We’ll look into some examples for you to reference once you get to that point of the work!
- Using the updated design language at this juncture is great! If you want more feedback as you progress the design, please let us know!

Looking forward to seeing more on this, great start Colleen!

**Reply by ceyerkes** [reply]
Just closing the loop! I think these are in a good spot - middle state is what i'm thinking would be what the cards look like mid-drag (if we even need that state?). I'm going to hold off on using the updated design language per @cfree, since engineering will likely start to work on these components before they're released.

![Screenshot 2023-01-23 at 9 08 25 AM](https://user-images.githubusercontent.com/87141551/214089213-714f1cd0-e1e2-4078-9427-db9a5e320b2d.png)

**Comment by ceyerkes** [comment]
Hello! Re-opening the loop 🫠

Even though the visual picker turned into it's own editor-specific component (drag & drop card) I did come across a use case for a visual picker in the image editor:

![Screenshot 2023-02-06 at 4 31 51 PM](https://user-images.githubusercontent.com/87141551/217110769-541f48c9-0af2-4964-b327-4c461540b660.png)

Let me know if you have any serious concerns, otherwise we'll proceed with this design, thanks!

**Reply by kelseycavitt** [reply]
@ceyerkes are you asking about the selected state of the filter in this example? I think we'd also need a design for the focus state and hover state (if they are different from the card)?
